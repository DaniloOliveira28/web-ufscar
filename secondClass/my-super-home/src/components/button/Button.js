import React from 'react';
import PropTypes from 'prop-types';
import './button.scss';

function Button(props) {
  const { type, onClick, alt, className } = props;
  return (
    <button
      type={type}
      alt={alt}
      className={className}
      onClick={onClick}
    />
  )
}

export default Button;

Button.defaultProps = {
  type: 'submit'
}

Button.propTypes = {
  alt: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  type: PropTypes.string,
  className: PropTypes.string.isRequired
}

