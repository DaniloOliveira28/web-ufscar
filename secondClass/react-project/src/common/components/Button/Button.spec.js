import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

const componentWrapper = ({ props }) => {
  return shallow(
    <Button
      {...props}
    />);
};

describe('Button component', () => {
  it('Should render button properly', () => {
    const props = {
      alt: 'alt', onClick: jest.fn(), className: 'button__search'
    };
    const button = componentWrapper({ props });
    expect(button).toMatchSnapshot();
  });
  it('Should be able to click', () => {
    const props = {
      alt: 'alt', onClick: jest.fn(), className: 'button__search'
    };
    const button = componentWrapper({ props });
    button.find('button').simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });
});


