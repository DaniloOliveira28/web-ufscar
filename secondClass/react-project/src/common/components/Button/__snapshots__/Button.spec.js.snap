// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should render button properly 1`] = `
<button
  alt="alt"
  className="button__search"
  onClick={[MockFunction]}
  type="submit"
/>
`;


